timesketch/lib/llms/providers/secgemini_log_analyzer_agent.py

```diff
@@ -14,168 +14,211 @@
+        """Initializes a SecGemini session and streams the analysis response.
 
-        This method is used internally to prepare data for the SecGemini API.
-        It converts each event dictionary to a JSON line and yields it as bytes.
+        This is an async helper method that:
+        1. Creates a new SecGemini session.
+        2. Uploads the local log file to the session.
+        3. Streams the analysis results for the given prompt.
 
         Args:
-            log_events_generator: An iterable of log event dictionaries.
+            log_path (Path): The local filesystem path to the JSONL log file.
+            prompt (str): The analysis prompt to send to the agent.
 
         Yields:
-            bytes: JSON-encoded event lines in NDJSON format.
+            str: The content chunks of the streamed response from the agent.
         """
-        logger.info("SecGemini Log Analyzer: Preparing to stream log data as NDJSON.")
-
-        for event_dict in log_events_generator:
-            try:
-                line_to_yield = (json.dumps(event_dict) + "\n").encode("utf-8")
-                yield line_to_yield
-                self._events_sent += 1
-            except TypeError as exception:
-                logger.error(
-                    "SecGemini Log Analyzer: Error serializing event to JSON: "
-                    "%s... Error: %s",
-                    str(event_dict)[:100],
-                    exception,
-                )
-            except Exception as exception:  # pylint: disable=broad-except
-                logger.error(
-                    "SecGemini Log Analyzer: Unexpected error during event "
-                    "serialization for stream: %s",
-                    exception,
-                    exc_info=False,
-                )
-
-        if self._events_sent == 0:
-            logger.warning(
-                "SecGemini Log Analyzer: _stream_log_data_as_ndjson_gen: "
-                "log_events_generator was empty or yielded no data."
+        self._session = self.sg_client.create_session(
+            model=self.model, enable_logging=self.enable_logging
+        )
+        self.session_id = self._session.id
+        logger.info("Started new SecGemini session: '%s'", self._session.id)
+        self._session.upload_and_attach_logs(
+            log_path, custom_fields_mapping=self.custom_fields_mapping
+        )
+        if hasattr(self._session, "logs_table") and hasattr(
+            self._session.logs_table, "blake2s"
+        ):
+            self.table_hash = self._session.logs_table.blake2s
+            logger.info(
+                "Uploaded logs table hash (blake2s): '%s'",
+                self._session.logs_table.blake2s,
             )
+        else:
+            logger.warning("Uploaded logs did not produce a blake2s table hash!")
+
+        logger.info("Starting the SecGemini analysis...")
+        async for response in self._session.stream(prompt):
+            yield response.content
```

Comment:
With multiple tests, I can't get this to work reliably. Sometimes it works and sometimes the websocket connection gets reset and I can't reliably reproduce it.

I'll look into options to make this more reliable.

Reply:
Trying to use `_session.query()` instead also runs into Timeout errors.

Reply:
Alright, decided to keep the streaming approach and let secgemini client handle the connection issues. For TS I went with some clear context info in case this happens.